.evergreen/init.sh

```diff
@@ -190,3 +190,44 @@ _init_sh_evg_dir="$(dirname "${_init_sh_this_file}")"
 # this one, and will therefore remain as native paths
 EVG_DIR="$(native_path "${_init_sh_evg_dir}")"
 LIBMONGOCRYPT_DIR="$(dirname "${EVG_DIR}")"
+
+# Executes CMake via the cache-managing script
+run_cmake() {
+    command bash "$EVG_DIR/cmake.sh" "$@"
+}
```

Comment:
```suggestion
run_cmake() (
    command bash "$EVG_DIR/cmake.sh" "$@"
)
```
Recommend using subshells `()` by default instead of scopes `{}` for new functions that do not require exporting variables to the environment.

Reply:
I haven't seen this idiom before. Is there a benefit to this shape? Also, would this subshell attempt to fork? On Windows the Cygwin fork() is abysmally slow.

Reply:
The benefit is explicitly documenting that the given block/function does not intend to modify its environment (self-contained). I was not aware that subshells in Cygwin were that slow compared to on Linux. I do not think it is severe enough to discourage use of subshells, but I may be mistaken.

Reply:
> does not intend to modify its environment (self-contained)

Worth noting that this also includes setting shell options such as `errexit`, `pipefail`, `nounset`, and `xtrace`. Manually preserving and restoring shell options [can be done](https://unix.stackexchange.com/a/310963), but it is complicated, unwieldy, and difficult to ensure all exit paths are reliably covered.

Reply:
Cygwin's fork() emulation is abysmally slow. It's on the order of dozens of milliseconds, which is "fast" to the eye, but adds up extremely quickly in a script execution. Not necessarily a problem here where a CMake subprocess will dominate the time, but I don't think the isolation of a subshell matters much here either, where the function is only running a single subcommand.